plugins/modules/host.py

```diff
@@ -484,6 +484,17 @@ def main():
         if not module.desired_absent:
             module.auto_lookup_entities()
 
+        if 'compute_resource' in module.foreman_params and 'interfaces_attributes' in module.foreman_params:
+            avn_params = {'id': module.foreman_params['compute_resource']['id']}
+            available_networks = module.resource_action('compute_resources', 'available_networks', params=avn_params,
+                                                        ignore_check_mode=True, record_change=False)['results']
+
+            for interface in module.foreman_params['interfaces_attributes']:
+                if 'compute_attributes' in interface and 'network' in interface['compute_attributes']:
+                    network_name = interface['compute_attributes']['network']
+                    network_id = next((network['id'] for network in available_networks if network['name'] == network_name), network_name)
+                    interface['compute_attributes']['network'] = network_id
```

Comment:
what should happen if the network was not found in the available ones? currently, this silently takes the value from the playbook, and will thus fail, I think

Reply:
I think, the best would be to print an error message, such as "network not found" or something